ui/src/components/admin/fundraise/AdminFundraiseGrid.js

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useEffect } from 'react';
+import { AgGridReact } from 'ag-grid-react';
+import 'ag-grid-community/dist/styles/ag-grid.css';
+import 'ag-grid-community/dist/styles/ag-theme-alpine.css';
+import { Heading, Container, Switch, HStack } from '@chakra-ui/react';
+
+export default function AdminFundraiseGrid(props) {
+  const { fundraises, setSelected } = props;
+
+  const [showDeleted, setShowDeleted] = useState(false);
+  const [filtered, setFiltered] = useState(fundraises);
+  const [api, setApi] = useState();
+  const [colApi, setColApi] = useState();
+
+  useEffect(() => {
+    if (fundraises) {
+      if (!showDeleted) {
+        setFiltered(fundraises.filter((p) => p.isDeleted === false));
+      } else {
+        setFiltered(fundraises);
+      }
+    }
+  }, [fundraises, showDeleted]);
+
+  useEffect(() => {
+    if (api) {
+      if (fundraises && fundraises.length) {
+        api.hideOverlay();
+      } else {
+        api.showLoadingOverlay();
+      }
+    }
+  }, [fundraises, api]);
+
+  // Ag-Grid Functions
+  const onGridReady = (params) => {
+    if (params.api) setApi(params.api);
+    if (params.columnApi) setColApi(params.columnApi);
+  };
+
+  const onRowClicked = ({ data }) => {
+    if (data) setSelected(data);
+  };
+
+  // Ag-Grid definitions
+  const columnDefs = [
+    { headerName: 'Campaign Name', field: 'campaignName' },
+    { headerName: 'Category Name', field: 'categoryName' },
+    { headerName: 'Category Key', field: 'categoryKey' },
+    { headerName: 'Amount', field: 'amount' },
+    { headerName: 'Givers', field: 'givers' },
+    { headerName: 'Milestones', field: 'milestones' },
```

Comment:
I think this is just nice to have for now, it feels kinda weird to see [object, object] in the ag-grid data columns lol

Reply:
resolved, displays full string of Milestones, not ideal but they can view the content when they click it anyways